DataServer/portAllocator.py

```diff
@@ -1,19 +1,40 @@
+import socket
+import contextlib
+
 class PortAllocator:
     """This class provides a method to allocate the next available port in the
     given block of ports.
     """
+
     def __init__(self, block_start: int, block_end: int) -> None:
         """Initializes the allocator with the specified block of ports
-
         Args:
             block_start (int): Starting number of block of ports, inclusive
             block_end (int): Ending number of block of ports, inclusive
         """
+
+        self.__s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
         self.__start = block_start
         self.__end = block_end
 
-        # TODO reimplement this
-        self.__current = self.__start
+        self.__reservedPorts = set()
+        self.__hostname = self.__s.gethostname()
+
+        self.__ip = self.__s.gethostbyname(self.__hostname)
```

Comment:
`gethostbyname` is not an instance member of `socket.socket`.  See https://docs.python.org/3.7/library/socket.html#socket.gethostbyname